Sources/firebase/include/FirebaseFirestore.hh

```diff
@@ -44,6 +44,37 @@ document_get(const ::firebase::firestore::DocumentReference document, ::firebase
   return document.Get(source);
 }
 
+inline ::firebase::firestore::DocumentReference
+snapshot_reference(const ::firebase::firestore::DocumentSnapshot snapshot) {
+  return snapshot.reference();
+}
+
+struct MapFieldValue_Workaround {
+  std::vector<std::string> keys;
+  std::vector<::firebase::firestore::FieldValue> values;
+};
+
+inline MapFieldValue_Workaround
+map_field_value_to_workaround(const ::firebase::firestore::MapFieldValue value) {
+  MapFieldValue_Workaround data;
+  data.keys.reserve(value.size());
+  data.values.reserve(value.size());
+
+  for (auto i = value.begin(); i != value.end(); i++) {
+    data.keys.emplace_back(i->first);
+    data.values.emplace_back(i->second);
+  }
```

Comment:
```suggestion
  for (const auto &kv : value) {
    data.keys.emplace_back(kv.first);
    data.values.emplace_back(kv.second);
  }
```

Reply:
Added this manually since I was making other changes.